src/components/signupPage/logic/signupLogic.jsx

```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export default function SignupLogic() {
+  const navigate = useNavigate();
+  const [signingIn, setSigningIn] = useState(false);
+  const [serverResponse, setServerResponse] = useState({
+    type: null,
+    text: null,
+  });
+
+  async function signup(data) {
+    if (!data.firstName || !data.lastName || !data.email || !data.password)
+      return setServerResponse({
+        type: "error",
+        text: "Kindly fill and the mandatory fields.",
```

Comment:
Found a typo here - "and" instead of "all". 
Also, on the carousel, it says "experince" instead of "experience". Please check later.